pkg/chunk/cache/fifo_cache.go

```diff
@@ -71,29 +74,40 @@ var (
 	}, []string{"cache"})
 )
 
+// This FIFO cache implementation supports two eviction methods - based on number of items in the cache, and based on memory usage.
+// For the memory-based eviction, set FifoCacheConfig.MaxSizeBytes to a positive integer, indicating upper limit of memory allocated by items in the cache.
+// Alternatively, set FifoCacheConfig.MaxSizeItems to a positive integer, indicating maximum number of items in the cache.
+// If both parameters are set, the memory-based eviction method takes precedence.
```

Comment:
If both parameters are set, cannot we enforce both limits, whichever is hit first? Why are they mutually exclusive?

Reply:
Indeed, we can support both.